engine/apps/alerts/escalation_snapshot/escalation_snapshot_mixin.py

```diff
@@ -232,17 +230,11 @@ def start_escalation_if_needed(self, countdown=START_ESCALATION_DELAY, eta=None)
 
         is_on_maintenace_or_debug_mode = self.channel.maintenance_mode is not None
 
-        if (
-            self.is_restricted
-            or is_on_maintenace_or_debug_mode
-            or self.pause_escalation
```

Comment:
Escalations for paused alert groups should not be skipped. Paused alert groups should be escalated further and get handled by the "Continue escalation if >X alerts per Y minutes" step.